tests/test_telemetry_monitor.py

```diff
@@ -400,10 +450,59 @@ def is_alive(self) -> bool:
 
             return True
 
-    job_entity1 = JobEntity()
-    job_entity1.name = "xyz"
-    job_entity1.step_id = "123"
-    job_entity1.task_id = ""
+    frequency = 1
+
+    monitor_pattern = f"{test_dir}/mock_mani.json"
+    # show that an event handler w/out a monitored task will automatically stop
+    mani_handler = ManifestEventHandler("xyz", logger)
+    observer = FauxObserver()
+    duration = 2
+
+    ts0 = get_ts_ms()
+    await event_loop(observer, mani_handler, frequency, duration)
+    ts1 = get_ts_ms()
+
+    assert ts1 - ts0 >= duration
+    assert observer.stop_count == 1
```

Comment:
buggy bug is buggy! I swapped to milliseconds at some point and missed a multiplication. You actually uncovered a more annoying bug - inside the event loop, i was basically running a no-op instead of a sleep due to sleeping the "seconds value" instead of the number of ms (e.g. 2ms instead of 2000).

great catch!

fixed here:
`wait_ms = max(1000 * frequency - duration_ms, 0)`